src/librustc/diagnostics.rs

```diff
@@ -279,6 +279,41 @@ println!("{}", Y);
 ```
 "##,
 
+E0019: r##"
+A function call isn't allowed in the const's initialization expression
+because the expression's value must be known at compile-time. Bad example:
+
+```
+enum Test {
+    V1
+}
+
+impl Test {
+    fn test(&self) -> i32 {
+        12
+    }
+}
+
+fn main() {
+    const FOO: Test = Test::V1;
+
+    const A: i32 = FOO.test(); // You can't call Test::func() here !
+}
+```
+
+Remember: you can't use a function call inside a const's initialization
```

Comment:
You should add an additional paragraph about the `const fn` feature, which is currently feature-gated and thus only available in Nightly (but there are other error explanations like E0152 that reference feature-gated things).  The sole purpose of `const fn` is *to be* a function call that breaks the rule you wrote here.  :)

Here is a demonstration of `const fn`:

  https://play.rust-lang.org/?gist=449fb78958b96eca7528&version=nightly